### PR TITLE
IDENTITY-6451: Fix relay state not returned to the SAML relying party

### PR DESCRIPTION
**Pull request by emswbandara**

### CODE REVIEW

**Comment by emswbandara** [comment]
@omindu @harsha1979 @danushkaf

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java [hunk 1]
```diff
@@ -305,6 +305,15 @@ private void sendNotification(String errorResp, String status, String message,
         if (acUrl != null) {
             queryParams += "&" + SAMLSSOConstants.ASSRTN_CONSUMER_URL + "=" + URLEncoder.encode(acUrl, "UTF-8");
         }
+        String relayState;
```
Comment (rushmin) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: Isn't below code has better performance and readability?

String relayState = req.getParameter(SAMLSSOConstants.RELAY_STATE);

// If the request doesn't have a relay state, get it from the session.
        if (StringUtils.isEmpty(relayState)) {
            relayState = sessionDTO.getRelayState();
        }

Comment (GayanM) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: String relayState = req.getParameter(SAMLSSOConstants.RELAY_STATE);
if(StringUtils.isEmpty(relayState){
    relayState = sessionDTO.getRelayState();
}

Comment (emswbandara) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java]: Fixed with 49929bb

**Pull request merged**